grids/SimilarityActionGridColumn.inc.php

```diff
@@ -0,0 +1,338 @@
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SimilarityActionGridColumn
+ * @ingroup plugins_generic_plagiarism
+ *
+ * @brief GridColumn handler to show similarity score and actions related to iThenticate
+ */
+
+import('lib.pkp.classes.db.DAORegistry');
+import('lib.pkp.classes.controllers.grid.GridColumn');
+import('lib.pkp.classes.controllers.grid.ColumnBasedGridCellProvider');
+import('lib.pkp.classes.linkAction.request.OpenWindowAction');
+import('lib.pkp.classes.linkAction.request.RemoteActionConfirmationModal');
+import('lib.pkp.classes.linkAction.request.AjaxModal');
+
+class SimilarityActionGridColumn extends GridColumn {
+
+	public const SIMILARITY_ACTION_GRID_COLUMN_ID = 'score';
+	
+	/** 
+	 * The Plagiarism Plugin itself
+	 * 
+	 * @var PlagiarismPlugin
+	 */
+	protected $_plugin;
+
+	/**
+	 * Constructor
+	 *
+	 * @param PlagiarismPlugin  $plugin The Plagiarism Plugin itself
+	 */
+    public function __construct($plugin) {
+
+		$this->_plugin = $plugin;
+
+		$cellProvider = new ColumnBasedGridCellProvider();
+
+		parent::__construct(
+			self::SIMILARITY_ACTION_GRID_COLUMN_ID,
+			'plugins.generic.plagiarism.similarity.action.column.score.title',
+			null,
+			null, 
+			$cellProvider,
+			['width' => 30, 'alignment' => COLUMN_ALIGNMENT_LEFT, 'anyhtml' => true]
+		);
+	}
+
+	/**
+	 * Method expected by ColumnBasedGridCellProvider to render a cell in this column.
+	 *
+	 * @copydoc ColumnBasedGridCellProvider::getTemplateVarsFromRowColumn()
+	 */
+	public function getTemplateVarsFromRow($row) {
+
+		if ($this->_plugin::isOPS()) { // For OPS
+			$articleGalley = $row->getData(); /** @var ArticleGalley $articleGalley */
+			
+			if (!isset($articleGalley->_data['submissionFileId'])) {
+				return ['label' => ''];
+			}
+
+			$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /** @var SubmissionFileDAO $submissionFileDao */
+			$submissionFile = $submissionFileDao->getById($articleGalley->getData('submissionFileId')); /** @var SubmissionFile $submissionFile */
+
+			$articleGalleyDao = DAORegistry::getDAO('ArticleGalleyDAO'); /** @var ArticleGalleyDAO $articleGalleyDao */
+			$articleGalley = $articleGalleyDao->getByFileId($submissionFile->getId()); /** @var ArticleGalley $articleGalley */
+		} else {
+			$submissionFileData = $row->getData();
+			$submissionFile = $submissionFileData['submissionFile']; /** @var SubmissionFile $submissionFile */			
+		}
+
+		assert($submissionFile instanceof SubmissionFile);
+
+		// Not going to allow plagiarims action on a zip file
+		if ($this->isSubmissionFileTypeZip($submissionFile)) {
+			return ['label' => __('plugins.generic.plagiarism.similarity.action.invalidFileType')];
+		}
+
+        // submission similarity score is available
+        if ($submissionFile->getData('ithenticateSimilarityScheduled') === true &&
```

Comment:
I think this strict `===` comparison will get you into trouble; it's a boolean test here, but when you set it for storage, [it's an integer 1](https://github.com/pkp/plagiarism/pull/57/files#diff-ca39d1234625667f81ff04dbca509cdb6ec27b5180b7595f9ca5a5b4e6ee8101R111). Best to be consistent with `true` / `false` vs. `1` and `0` (and I prefer booleans), but also not to run strict comparisons.